scripts.js

```diff
@@ -0,0 +1,98 @@
+window.onload = () => {
+  function isValidData(data, pattern) {
+  return pattern.test(data);
+}
+
+
+document.querySelector('#submitButton').disabled = true;
+
+const changeButtonState = function() {
+  if ( 
+    ((isNameEntered && isNumberEntered) || (isCheckboxChecked && isNameEntered)
```

Comment:
you use those properties before declare. Do not do it, pass as function arguments or declare function later

It works because of the dynamic variables scope https://github.com/azat-io/you-dont-know-js-ru/blob/master/scope%20%26%20closures/apA.md

But it is very bad practice, because of hard of debug and you cannot reuse this code